### PR TITLE
Allow building without throwing exceptions internally

### PR DESCRIPTION
**Pull request by MJansson**
Make exceptions options with build flag (default on).
Separate debug/info log from warn/error log
Allow exceptions to be cleared in checkException
Refactor in ref classes to avoid env calls when not needed

### CODE REVIEW

File: src/main/cpp/JavaExceptionUtils.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ class JavaExceptionUtils {
 public:
   static EXPORT JniLocalRef<jobject> newThrowable(JNIEnv *env, const char *message, ...);
 
-  static EXPORT void checkException(JNIEnv *env);
+  static EXPORT void checkException(JNIEnv *env, bool clear = false);
```
Comment (nikreiman) [inline src/main/cpp/JavaExceptionUtils.h]: I'm not a huge fan of default arguments; it makes it difficult if we want to add another parameter to this method etc. Could we refactor to have 2 methods instead? I would propose calling it `checkExceptionAndClear()` and just skip the bool parameter.

File: src/main/cpp/JniHelpersCommon.h [hunk 2]
```diff
@@ -63,23 +63,42 @@
...
+#endif
+
 // Determines whether logging messages will be printed by JniHelpers
 #ifndef ENABLE_LOGGING
 #define ENABLE_LOGGING 0
 #endif
 
+// Determines whether debug/info logging messages will be printed by JniHelpers
+#ifndef ENABLE_LOGGING_DEBUG
+#define ENABLE_LOGGING_DEBUG 0
```
Comment (nikreiman) [inline src/main/cpp/JniHelpersCommon.h]: Given the name of this symbol, it should probably be defined to `= DEBUG` by default

**Comment by nikreiman** [comment]
Overall the change looks good, just a few minor style notes as commented above.

**Comment by MJansson** [comment]
Good suggestions, updated the pull requests

**Comment by nikreiman** [comment]
Since I only have one small nitpick on this review I'm just gonna merge the PR and fix it with the inline GH editor. Thanks for the fix! :)

**Pull request merged**